### ISSUE TITLE
Discrepancy in Special Token Usage for Qwen3 Embedding Model

### ISSUE THREAD

**Issue by wintertee** [QwenLM/Qwen3-Embedding#122]
I've noticed a potential discrepancy between the official paper and the code implementation regarding the special token appended to the input for the Qwen3 embedding model.

Paper: The technical report states that the End-of-Sentence (`EOS`) token should be appended to the input sequence.

<img width="366" height="310" alt="Image" src="https://github.com/user-attachments/assets/3e00d6b4-31ab-499b-84ab-1b0380ce74c1" />

Code Implementation: However, the provided code implementation automatically appends the Padding (`PAD`) token to the end of the input sequence.

```python
tokenizer = AutoTokenizer.from_pretrained('Qwen/Qwen3-Embedding-0.6B', padding_side='left')
pprint(tokenizer(["test", "test again"],padding=True))
pprint(f"{tokenizer.eos_token_id=},{tokenizer.pad_token_id=}")
```

output:
```python
{'attention_mask': [[0, 1, 1], [1, 1, 1]],
 'input_ids': [[151643, 1944, 151643], [1944, 1549, 151643]]}
'tokenizer.eos_token_id=151645,tokenizer.pad_token_id=151643'
```

As shown in the output, the `PAD` token (151643) is used for padding and is present at the end of the shorter sequence, while the paper suggests using the `EOS` token (151645).

Could you please clarify which token is the correct one to use at the end of a sequence for optimal performance? If the use of the PAD token is intentional, could you please explain the reasoning behind this choice?

Thank you for your great work on this project!

**Comment by wintertee** [comment]
Thank you very much for your explanation!

**Comment by Oukaishen** [comment]
大佬你好，我们最近有需求使用emb模型 + cls分类头来做一个线上分类。

具体遇到问题就是，eos和pod都是151643，这样的话right pad情况下，HF训练的时候都是拿最后一个非pad, 意味着是eos前面一个（eos也被跳过了）

只能用left pad训练，请教一下这个理解对吗
